### PR TITLE
br: add res.Body.close to avoid leak

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

br: add res.Body.close to avoid leak

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/split_client.go [hunk 1]
```diff
@@ -416,6 +417,7 @@ func (c *pdClient) getMaxReplica(ctx context.Context) (int, error) {
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
+	defer terror.Log(res.Body.Close())
```
Comment (hawkingrei) [inline br/pkg/restore/split_client.go:420]: It is the right way to log ?

**Review by YuJuncen — APPROVED** [review]

**Review by kennytm — APPROVED** [review]

**Pull request merged**